examples/list_red_moving_vehicle/yolox_detector.py

```diff
@@ -0,0 +1,92 @@
+"""
+Based on demo implementation in Megvii YOLOX repo
+The YOLOX detector for object detection
+"""
+
+from typing import Dict, List
+
+import numpy as np
+import torch
+from loguru import logger
+from vqpy.base.detector import DetectorBase
+from vqpy.utils.classes import COCO_CLASSES
+from vqpy.detector.logger import register
+
+from yolox.data.data_augment import ValTransform
+from yolox.exp.build import get_exp
+from yolox.utils import postprocess
+from yolox.utils.model_utils import get_model_info
+
+
+def generate_register_yolox_detector(model_path=None):
+
+    class YOLOXDetector(DetectorBase):
```

Comment:
Is there any specific reason to define `YOLOXDetector` as an inner class? Inner class code looks a little bit difficult to understand.

Reply:
It is for users to ingest `model_path` for their pretrained model (without changing the Detector interface).

I have updated code. In the newer version, users can specify a `detector_model_path` in `vqpy.launch`, and `model_path` is an instance attribute of Detector.  
```python
class YOLOXDetector(DetectorBase):

    def __init__(self, model_path, device="gpu", fp16=True):
          pass 

vqpy.launch(...,
     detector_model_path=args.yolox_model_path,
)
```
How about the updated interface?

Reply:
Looks better